src/app/map/map.component.ts

```diff
@@ -74,12 +80,27 @@ export class MapComponent implements AfterViewInit {
     this.updateStyles();
   }
 
-  constructor(private _ngZone: NgZone) {
+  constructor(private _ngZone: NgZone, private iterableDiffers: IterableDiffers) {
+    this._iterableDiffer = iterableDiffers.find([]).create(null);
     this.pendingStyles = fetch('assets/basemap.json', {credentials: 'include'})
       .then((response) => response.json());
   }
 
-  /**
+  ngDoCheck() {
+    let changes = this._iterableDiffer.diff(this._rows);      
+    if (changes) {
+      this._numChanges++;     
+        if (this._numChanges >= this._batchSize) {
+         this.updateFeatures();
+         this.updateStyles();
+         this._numChanges = 0;
+	 // Increase the batch size incrementally to keep the overhead low.
+         this._batchSize = this._batchSize * 1.5;
+      }
+    } 
+  }
```

Comment:
What would be the effect if we pulled the diffing changes out of this PR, and discuss them separately? 

I'm assuming pagination still works up to a point, but somewhere (well under 250MB I'm sure) the map updates get slow?

I think there are other ways to do this more simply and efficiently, but I don't necessarily want to slow down this PR with exploring them.

Reply:
I think we should definitely improve this if you have ideas. I have some ideas myself. I tried a bunch of them based on the tutorial in deck.gl but didn't make much different. We should discuss your ideas. I'll start a thread with you around this. 

The way this is written right now, even with 250MB things are not that bad. At 250MB-ish the map update takes 5 seconds which is not terrible.

If I remove this from here and don't batch the updates, end to end the query is much slower. I think regardless of what other changes we make, we should do the batching. If someone is showing 250MB of data on a map, they don't need to see the iterative progress for every 10MB anyways.

Reply:
I don't want to make experimental performance suggestions in this PR, since it appears to block the Firebase PR, let's get this PR finished and then we can improve performance in a PR that is fully separated from the Firebase change.

My biggest concern is that `_batchSize` never goes down. If the user issues large queries repeatedly, does that eventually become a problem?

(aside: this file also needs to be reformatted it looks like?)

Reply:
Each batch is basically on page of results. So, _batchsize cannot be more than 25.

Reply:
I looked at this again, I think we need to reset the batch size when one query is done....

Reply:
Added code to reset the batch size to default 5 in the begging of each query. Basically, every time the _rows are set, we assume a new query is started.